### ISSUE TITLE
Please support building the library on GNU Hurd and Debian kFreeBSD

### ISSUE THREAD

**Issue by copyninja** [creytiv/re#37]
Hi,

We see there is a build failure of re on the non GNU Linux ports provided by Debian, kFreeBSD and GNU Hurd. It would be good if upstream can provide support for building on these platform.

Here is build logs on these platform https://buildd.debian.org/status/package.php?p=libre

**Comment by alfredh** [comment]
the build log says this:

```
> /usr/bin/ld: build-i586/mod/dl.o: undefined reference to symbol 'dlclose@@GLIBC_2.2.6'
```

I am wondering what could be causing this..

**Comment by copyninja** [comment]
Hi Alfred,

I've fixed the build on `GNU Hurd` and `GNU kFreeBSD` the patch is available [here](https://anonscm.debian.org/cgit/pkg-voip/libre.git/plain/debian/patches/1001_support_hurd_kfreebsd.patch). After this patch now `libre` `librem` and `baresip` all builds fine on `GNU Hurd` and `GNU kFreeBSD`. Please consider merging the patch.

Also there is one more [patch](https://anonscm.debian.org/cgit/pkg-voip/libre.git/plain/debian/patches/0001-get-correct-multiarch-triplet.patch) fixing the multi-arch triplet generation. This was needed to make the package built with epoll support on Debian architectures (other than 386). If you can adopt it that would be great.

**Comment by alfredh** [comment]
could you attach the output of `make info` on both platforms?

with the above patch, do you know if https://github.com/creytiv/retest
runs successfully on both platforms?

**Comment by copyninja** [comment]
You can find make info output on these build logs
https://buildd.debian.org/status/package.php?p=libre

Just click on installed link infront of architecture name.

I can check retest on those platform if you want and provide any output
needed by you.

**Comment by alfredh** [comment]
fixed in commit 334ec7501106d31f4c86ee4173d3cf7c3eddfc72

thanks for the patch :)
